### PR TITLE
Updating the CONTRIBUTING file

### LINKED ISSUE #15: Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Anas-mhameed**
Adding a standard about PR Review Process
resolve #15

### CODE REVIEW

**Review by rawadabu — APPROVED** [review]

**Review by finziamit — APPROVED** [review]
LGTM

**Review by Shalevro2 — APPROVED** [review]

**Review by yaelnoyman1 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ After the installations are complete successfully, by typing 'vagrant up' in the
 3.	Make sure the PR has been checked thoroughly before creating the new PR. todo
```
Comment (Yarboa) [inline CONTRIBUTING.md]: ```... before creating the new PR. todo```

Did you forget to delete it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ After the installations are complete successfully, by typing 'vagrant up' in the
 3.	Make sure the PR has been checked thoroughly before creating the new PR. todo  
 4.	flake8 will test relevant python code.  
 5.	if the PR gets enough support from the maintainers, it will be approved and merged.
```
Comment (Yarboa) [inline CONTRIBUTING.md]: maintainers? 
Please replace the terms are Reviewers -> Contributors, and mentos -> Core Reviewers

**Review by Yarboa — CHANGES_REQUESTED** [review]
Consider Renaming the ## Pre-Requirements -> Dev environment setup

Consider adding guideline for quick CR(Core Reviewer) Review:

Commit description should include subject, empty line followed with comment in more details
In case commit resolves and Issue please resolves #Issue_id

All automation test should pass successfully check (flake8 is just one test from more to come)

PR should be as smaller as possible.

PR should not have merge conflicts (rebase, push --force)

PR should (optional) include tests which covers the PR change

At least 2 Reviewers should Approve this PR

I hopw it makes more sense now

File: CONTRIBUTING.md (hunk 1 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
   
 ## Pull Request Process  
 1.	Create an issue, address the Assignees and the Label.  
-2.	Clone the repository and make a contribution, commit change\s.  
-3.	Make sure the PR has been checked thoroughly before creating the new PR. todo  
+2.	Clone the repository and make a contribution, commit changes.  
+3.	Make sure the PR has been checked thoroughly before creating the new PR.
 4.	flake8 will test relevant python code.  
-5.	if the PR gets enough support from the maintainers, it will be approved and merged.  
+5.	if the PR gets enough support from the Reviewers(Contributors, and mentors), it will be approved and merged.
```
Comment (Yarboa) [inline CONTRIBUTING.md]: %s/mentors/Core Reviewers/

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,19 +1,29 @@
...
 ## Pull Request Process  
 1.	Create an issue, address the Assignees and the Label.  
-2.	Clone the repository and make a contribution, commit change\s.  
-3.	Make sure the PR has been checked thoroughly before creating the new PR. todo  
+2.	Clone the repository and make a contribution, commit changes.  
+3.	Make sure the PR has been checked thoroughly before creating the new PR.
 4.	flake8 will test relevant python code.  
-5.	if the PR gets enough support from the maintainers, it will be approved and merged.  
+5.	if the PR gets enough support from the Reviewers(Contributors, and mentors), it will be approved and merged.  
+*	note that each PR should get at least 2 members reviews and only then the mentors will do their code reviews.
```
Comment (Yarboa) [inline CONTRIBUTING.md]: Consider markdown bold option
please refer this
https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,19 +1,29 @@
...
 *	note that conflicts within the PR may cause the PR to be disapproved.
```
Comment (Yarboa) [inline CONTRIBUTING.md]: same as above

**Review by Yarboa — APPROVED** [review]

**Review by Yarboa — CHANGES_REQUESTED** [review]
Please go over latest comments

File: CONTRIBUTING.md (hunk 3 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: Can explain, please ??

Comment (Yarboa) [reply CONTRIBUTING.md]: Please change mentors -> Core Reviewers (CR)

Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 5 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,19 +1,29 @@
...
+2.	Clone the repository and make a contribution, commit changes.  
+3.	Make sure the PR has been checked thoroughly before creating the new PR.
 4.	flake8 will test relevant python code.  
-5.	if the PR gets enough support from the maintainers, it will be approved and merged.  
-*	note that conflicts within the PR may cause the PR to be disapproved.  
+5.	if the PR gets enough support from the Core Reviewers(CR), it will be approved and merged.  
+	* **each PR should get at least 2 members reviews and only then the mentors will do their code reviews.**
+	* **conflicts within the PR may cause the PR to be disapproved.**  
+
+## CR(Core Reviewer) Review
```
Comment (Yarboa) [inline CONTRIBUTING.md:19]: Please consider this

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,19 +1,29 @@
...
 ## Pull Request Process  
 1.	Create an issue, address the Assignees and the Label.  
-2.	Clone the repository and make a contribution, commit change\s.  
-3.	Make sure the PR has been checked thoroughly before creating the new PR. todo  
+2.	Clone the repository and make a contribution, commit changes.  
+3.	Make sure the PR has been checked thoroughly before creating the new PR.
 4.	flake8 will test relevant python code.  
-5.	if the PR gets enough support from the maintainers, it will be approved and merged.  
-*	note that conflicts within the PR may cause the PR to be disapproved.  
+5.	if the PR gets enough support from the Core Reviewers(CR), it will be approved and merged.
```
Comment (Yarboa) [inline CONTRIBUTING.md]: Please note the difference between Reviewers and Core Reviewers

Consider this suggestion
5. Once PR gets enough approvals from the Reviewers, 
It is ready to merge,
6. Merge is done by Core Reviewers once it is approved by Reviewers.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,19 +1,29 @@
...
 1.	Create an issue, address the Assignees and the Label.  
-2.	Clone the repository and make a contribution, commit change\s.  
-3.	Make sure the PR has been checked thoroughly before creating the new PR. todo  
+2.	Clone the repository and make a contribution, commit changes.  
+3.	Make sure the PR has been checked thoroughly before creating the new PR.
 4.	flake8 will test relevant python code.  
-5.	if the PR gets enough support from the maintainers, it will be approved and merged.  
-*	note that conflicts within the PR may cause the PR to be disapproved.  
+5.	if the PR gets enough support from the Core Reviewers(CR), it will be approved and merged.  
+	* **each PR should get at least 2 members reviews and only then the mentors will do their code reviews.**
```
Comment (Yarboa) [inline CONTRIBUTING.md]: I think you can remove this one

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,19 +1,29 @@
...
+	* **conflicts within the PR may cause the PR to be disapproved.**
```
Comment (Yarboa) [inline CONTRIBUTING.md]: I think you can remove this one also, it is covered in next section

File: CONTRIBUTING.md (hunk 6 above)
Comment (Yarboa) [inline CONTRIBUTING.md]: Please consider this:

Guidelines for quick PR merge

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,19 +1,29 @@
...
-5.	if the PR gets enough support from the maintainers, it will be approved and merged.  
-*	note that conflicts within the PR may cause the PR to be disapproved.  
+5.	if the PR gets enough support from the Core Reviewers(CR), it will be approved and merged.  
+	* **each PR should get at least 2 members reviews and only then the mentors will do their code reviews.**
+	* **conflicts within the PR may cause the PR to be disapproved.**  
+
+## CR(Core Reviewer) Review
+* Commit description should include subject, empty line followed with comment in more details
+* In case commit resolves and Issue please resolves #Issue_id
+* All automation test should pass successfully check (flake8 is just one test from more to come)
```
Comment (Yarboa) [inline CONTRIBUTING.md]: I think it is redundant

**Review by Yarboa — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 10 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 6 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 9 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 8 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 7 above)
Comment (Anas-mhameed) [reply CONTRIBUTING.md]: done

**Review by rawadabu — APPROVED** [review]
LGTM

**Review by Shalevro2 — APPROVED** [review]

**Review by Yarboa — APPROVED** [review]

**Pull request merged**